iota/commands/extended/promote_transaction.py

```diff
@@ -0,0 +1,68 @@
+# coding=utf-8
+from __future__ import absolute_import, division, print_function, \
+  unicode_literals
+
+import filters as f
+from iota import (
+  Bundle, TransactionHash, Address, ProposedTransaction, BadApiResponse,
+)
+from iota.commands import FilterCommand, RequestFilter
+from iota.commands.core.check_consistency import CheckConsistencyCommand
+from iota.commands.extended.send_transfer import SendTransferCommand
+from iota.filters import Trytes
+
+__all__ = [
+  'PromoteTransactionCommand',
+]
+
+
+class PromoteTransactionCommand(FilterCommand):
+  """
+  Executes ``promoteTransaction`` extended API command.
+
+  See :py:meth:`iota.api.Iota.promote_transaction` for more information.
+  """
+  command = 'promoteTransaction'
+
+  def get_request_filter(self):
+    return PromoteTransactionRequestFilter()
+
+  def get_response_filter(self):
+    pass
+
+  def _execute(self, request):
+    depth                 = request['depth'] # type: int
+    min_weight_magnitude  = request['minWeightMagnitude'] # type: int
+    transaction           = request['transaction'] # type: TransactionHash
+
+    cc_response = CheckConsistencyCommand(self.adapter)(tails=[transaction])
```

Comment:
I wanted to use the new `is_promotable` helper here, but we don't have a reference to the `Iota` class. I don't think this is a problem as it doesn't cause much duplication, but if we have more helper functions that we want to use from these commands in the future, it might need some thought on how to achieve that.

Reply:
Hm... interesting.

What do you think about this?

```python
class Iota:
  @property
  def helpers(self):
    return Helpers(self.adapter)
```

Then `PromoteTransactionCommand` can do `Helpers(self.adapter).is_promotable(...)`.

Reply:
Does that mean that the current [`is_promotable`](https://github.com/iotaledger/iota.lib.py/blob/develop/iota/commands/extended/helpers.py#L10) should be changed from:

```python
  def is_promotable(self, tail):
    return self.api.check_consistency(tails=[tail])['state']
```

To:
```python
  from iota.commands.core.check_consistency import CheckConsistencyCommand

  def is_promotable(self, tail):
    return CheckConsistencyCommand(self.adapter)(tails=[tail])['state']
```

I don't see a problem with that.

Reply:
Aye, I think that would... oh.  Hm, we might end up with circular references that way.

Eh, I guess we could import the commands locally in the helper methods.  The alternative, I think, would be to create helper classes, similar to what we're doing for commands.  That feels a bit cathedral-ish to me, though.

Reply:
Yeah, I can foresee circular dependencies. I'm okay with keeping it as is (not using the helper call) for the moment until a later date.